src/main/java/http/routes/structured_data/XMLResponse.java

```diff
@@ -0,0 +1,34 @@
+package http.routes.structured_data;
+
+import http.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class XMLResponse implements Route {
+    final static String CRLF = "\r\n";
+    final List<String> headers;
+    final String headersResponse;
+    final String xmlBody;
+
+    public XMLResponse() {
+        headers = Arrays.asList(Methods.GET.getMethod(), Methods.HEAD.getMethod());
+        headersResponse = String.format("%s%sAllow: %s, %s", ContentTypes.CONTENT_TYPE_XML.getType(), CRLF, headers.get(0), headers.get(1));
+        xmlBody = "<note><body>XML Response</body></note>";
+    }
+    @Override
+    public Response response(String method, String body) {
+        if (headers.contains(method)) {
+            return new ResponseBuilder()
+                    .withStatusCode(HTTPStatusCodes._200.getCode())
+                    .withHeader(headersResponse)
+                    .withBody(xmlBody)
+                    .build();
+        }
+        return new ResponseBuilder()
```

Comment:
This is a great opportunity to refactor and remove duplicate code here, as this return statement is at least 9 places: 
``` return new ResponseBuilder()
                .withStatusCode(HTTPStatusCodes._405.getCode())
                .withHeader(headersResponse)
                .withBody("")
                .build();
```

It is being called when there is a target route but not the correct method. Hint: I would look into adapting the `Router.getResponse` method to route to a `MethodNotFound` route.

Reply:
Addressed in the refactor.